.github/workflows/run_integration_tests.yml

```diff
@@ -2,13 +2,406 @@ name: Run Integration Tests
+          else
+            # Just use the supplied list of namespaces
+            namespaces="[ ${{ github.event.inputs.unbounded-test-namespaces }} ]"
+          fi
+          echo "::set-output name=matrix::$namespaces"
+        shell: bash
+
+
+
+  run-integration-tests:
+    needs: [setup-matrices]
+    if: github.event.inputs.solutions == 'Both' || github.event.inputs.solutions == 'IntegrationTests'
+    name: Run IntegrationTests
+    runs-on: windows-2019 # TODO: make this a input so we can test different OSes?
+    strategy:
+      matrix:
+        namespace: ${{fromJSON(needs.setup-matrices.outputs.integration-tests-matrix)}}
+      fail-fast: false # we don't want one test failure in one namespace to kill the other runs
+
+    env:
+      integration_tests_shared_project: ${{ github.workspace }}/tests/Agent/IntegrationTests/Shared
+      xunit_console: ${{ github.workspace }}/build/Tools/XUnit-Console/xunit.console.exe
+      integration_tests_dll: ${{ github.workspace }}/tests/Agent/IntegrationTests/IntegrationTests/bin/Release/net462/NewRelic.Agent.IntegrationTests.dll
+      # Make this variable true to enable extra data-gathering and logging to help troubleshoot test failures, at the cost of additional time and resources
+      enhanced_logging: false
+      NR_DOTNET_TEST_SAVE_WORKING_DIRECTORY : 1
+      
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup .NET Core 2.2.103
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '2.2.103'
+
+      - name: Setup .NET Core 3.1.100
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.100'
+
+      - name: Setup .NET 5.0
+        uses: actions/setup-dotnet@v1
+        with:
+            dotnet-version: '5.0.100'
+
+      - name: Setup .NET 6.0
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.x'
+          include-prerelease: true
+
+      - name: Set up secrets
+        env:
+          INTEGRATION_TEST_SECRETS: ${{ secrets.TEST_SECRETS }}
+        run: |
+          "$Env:INTEGRATION_TEST_SECRETS" | dotnet user-secrets set --project ${{ env.integration_tests_shared_project }}
+        shell: pwsh #this doesn't work with normal powershell due to UTF-8 BOM handling
+
+      - name: Download Agent Home Folders
+        uses: actions/download-artifact@v2
+        with:
+          name: homefolders
+          path: src/Agent
+
+      - name: Download Integration Test Artifacts
+        uses: actions/download-artifact/@v2
+        with:
+          name: integrationtests
+          # Should not need a path because the integration test artifacts are archived with the full directory structure
+
+      - name: Install dependencies
+        run: |
+          Enable-WindowsOptionalFeature -Online -FeatureName IIS-HostableWebCore
```

Comment:
The tag used the XUnit traits mechanism.  It would be nice if tests that required use of HWC were isolated to a few namespaces and then we could conditionally install HWC based on that.  We currently have a lot of tests that use a .NET framework web app (which generally means using HWC) that aren't really testing web instrumentation and could be refactored to use a console app instead, and this could make it easier to isolate the namespaces that require HWC.

Reply:
Yeah, maybe we could figure something out in the future, but I can see how it isn't easy to determine... In the future we might want to try removing this step and seeing what fails to get the list of what needs it 🤷‍♂️